src/functionalTest/groovy/com/autonomousapps/android/TestFixturesSpec.groovy

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024. Tony Robalik.
+// SPDX-License-Identifier: Apache-2.0
+package com.autonomousapps.android
+
+import com.autonomousapps.android.projects.TestFixturesProject
+import org.gradle.util.GradleVersion
+
+import static com.autonomousapps.advice.truth.BuildHealthSubject.buildHealth
+import static com.autonomousapps.utils.Runner.build
+import static com.google.common.truth.Truth.assertAbout
+
+final class TestFixturesSpec extends AbstractAndroidSpec {
+
+  def "should not falsely report duplicated dependencies with main source set(#gradleVersion AGP #agpVersion)"() {
```

Comment:
I feel like we could use a few more tests to have confidence the new feature works. Maybe validate that we correctly report on unused dependencies, used transitive dependencies, and incorrect configurations.